modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy

```diff
@@ -84,8 +85,7 @@ class BashWrapperBuilder {
         catch( Exception e ) {
             log.warn "Invalid value for `NXF_DEBUG` variable: $str -- See http://www.nextflow.io/docs/latest/config.html#environment-variables"
         }
-        BASH = Collections.unmodifiableList( level > 0 ? ['/bin/bash','-uex'] : ['/bin/bash','-ue'] )
-
+        BASH = Collections.unmodifiableList( level > 0 ? [NF.bash(), '-uex'] : [NF.bash(), '-ue'] )
```

Comment:
This use of `NF.bash()` makes me think that the user bash setting can only be the bash path itself and can't include extra arguments like `env -S bash` or `bash -o pipefail`. Will that be acceptable?

Reply:
The goal should be to specify a custom Bash path not to alter the shell option. That's possible in some extent via `process.shell`